config/config.go

```diff
@@ -264,7 +297,21 @@ func valueToYaml(value string) (*yaml.Node, error) {
 	}
 	fixStyling(valueNode)
 
-	return valueNode.Content[0], nil
+	contentNode := &yaml.Node{}
+	switch {
```

Comment:
nit: given that we aren't using the switch with a single value, I think using a if/elseif/else is more readable.

Reply:
Since they don't require a single value like other languages, I've personally found switch statements in Go to be more readable than a series of if/else-if/else, and have tended to use them liberally.

Using switch:
```golang
	switch {
	case args[1] == "--delete":
		outConfig, err = config.Delete(inConfig, args[0])
		if err != nil {
			return errors.Wrap(err, "delete config")
		}
	case args[1] == "--help" || args[1] == "-h":
		if err = config.PrintHelp(args[0]); err != nil {
			return errors.Wrap(err, "help")
		}
		return nil // exit now without writing any changes to config
	default: // args are key/value pairs, e.g. ["global.conversion_parallelism","5"]
		outConfig, err = config.Upsert(inConfig, args[0], args[1])
		if err != nil {
			return errors.Wrap(err, "upsert config")
		}
	}
```
Using if:
```golang
	if args[1] == "--delete" {
		outConfig, err = config.Delete(inConfig, args[0])
		if err != nil {
			return errors.Wrap(err, "delete config")
		}
	} else if args[1] == "--help" || args[1] == "-h" {
		if err = config.PrintHelp(args[0]); err != nil {
			return errors.Wrap(err, "help")
		}
		return nil // exit now without writing any changes to config
	} else { // args are key/value pairs, e.g. ["global.conversion_parallelism","5"]
		outConfig, err = config.Upsert(inConfig, args[0], args[1])
		if err != nil {
			return errors.Wrap(err, "upsert config")
		}
	}
```

Maybe it's just personal taste but I find the switch statement looks a bit cleaner. Happy to change it though if we have that preference for using if's in these cases throughout the project already though

Reply:
I like how it was used in 
```
	// Enable version-specific features.
	switch {
	case versionAtLeast(ftrs, 0, 6):
		ftrs.ReferencedSaveOnly = true
		ftrs.UseCopyIncludePatterns = true
		ftrs.ForIn = true
		ftrs.RequireForceForUnsafeSaves = true
		ftrs.NoImplicitIgnore = true
	case versionAtLeast(ftrs, 0, 7):
		ftrs.EarthlyVersionArg = true
	}
```

and in the example you shared, it can further be reduced into `switch args[1] {` (which then switches work great for).

In both cases the switch is centred around different cases for a single condition (major/minor feature, or arg).

I think the part that jumped out at me is that `valueNode.Content` and `value` aren't as tightly connected to a single variable, or even variable type.

For this discussion, it might be more useful to include the code that I was commenting on:

```
	switch {
	case len(valueNode.Content) > 0:
		// ContentNode contains the user-provided value with it's type etc
		contentNode = valueNode.Content[0]
	case value == "":
		// Edge case where the yaml.Unmarshal above results in no nodes in valueNode.Content.
		// The code below ensures we can write an actual empty string to our yaml as requested.
		contentNode.SetString("")
	default:
		// Very unlikely
		return nil, errors.New("failed setting value in yaml")
	}
```

vs

```
	if len(valueNode.Content) > 0 {
		// ContentNode contains the user-provided value with it's type etc
		contentNode = valueNode.Content[0]
	} else if value == "" {
		// Edge case where the yaml.Unmarshal above results in no nodes in valueNode.Content.
		// The code below ensures we can write an actual empty string to our yaml as requested.
		contentNode.SetString("")
	} else {
		// Very unlikely
		return nil, errors.New("failed setting value in yaml")
	}
```

Maybe @andrewsykim or @vladaionescu have thoughts on adopting this coding style?

Reply:
It's a nit, so any option is fine TBH. In any case, I think I like if/elseif/else when the conditions are on different variables and switch when the conditions are about the same thing.

Reply:
Yeah I totally pasted the wrong example 😅 

The example I did paste though might be worse as `switch args[1]` as we may have to duplicate the code for the --help case. e.g.
```golang
switch args[1]:
case "-h"
 // help
case "--help"
 // help again
case "--delete"
```
So I think I would prefer keeping as switch without arg or just use if/else.

Anyway, will patch both to use ifs instead

Reply:
You can still use switch for that https://www.geeksforgeeks.org/golang-program-that-uses-switch-multiple-value-cases/